influxql/parser.go

```diff
@@ -1739,7 +1817,18 @@ func (p *Parser) parseUnaryExpr() (Expr, error) {
 }
 
 // parseRegex parses a regular expression.
-func (p *Parser) parseRegex() (Expr, error) {
+func (p *Parser) parseRegex() (*RegexLiteral, error) {
+	nextRune := p.peekRune()
+	if isWhitespace(nextRune) {
```

Comment:
Do you have to conditionally consume the whitespace? I thought we had a function that just did, whether there was whitespace or not.